Av/caption gen clean

ashvarghese opened the pull request:
clean branch w resolved merge conflicts i believe

ashvarghese committed:
feat: Add caption generation endpoint and integration

ashvarghese committed:
feat: Add caption generation endpoint and integration

ashvarghese committed:
feat: Merge caption generation with main branch functionality

ashvarghese committed:
feat: Merge caption generation with main branch functionality

ashvarghese committed:
fix: Resolve merge conflicts with main

Kylejeong2 commented on apps/backend/requirements.txt:
```diff
@@ -1,9 +1,7 @@
+boto3==1.34.69
 flask-sqlalchemy==3.1.1
-pymysql==1.1.1
-runwayml==2.3.4
```
dont we need runway ml to run this?

Kylejeong2 commented on apps/backend/app.py:
```diff
@@ -43,32 +43,64 @@ class Video(db.Model):
+        })
+        
     except Exception as e:
-        db.session.rollback()  
```
not sure if we should delete this @kvijay3 is this important

kvijay3 reviewed (changes requested):
remove route make functions import and only change generate-videos route

ashvarghese committed:
removed route made functions import and changed generate-videos route

ashvarghese committed:
integrate caption generation into generate_videos function with default of 3 captions

ashvarghese committed:
Removed standalone generate-captions route

ashvarghese committed:
Resolve merge conflicts

ashvarghese commented:
Integrated caption generation into the generate_videos function
Set the default number of captions to 3
Added proper error handling
Removed the standalone /api/generate-captions route
Made the code compatible with different versions of the OpenAI library
